examples/js/loaders/OBJLoader2.js

```diff
@@ -0,0 +1,996 @@
+/**
+  * @author Kai Salmen / https://kaisalmen.de
+  * Development repository: https://github.com/kaisalmen/WWOBJLoader
+  */
+
+'use strict';
+
+if ( THREE.OBJLoader2 === undefined ) { THREE.OBJLoader2 = {} }
+THREE.OBJLoader2.version = '1.0.1';
+
+/**
+ * Use this class to load OBJ data from files or to parse OBJ data from arraybuffer or text
+ * @class
+ *
+ * @param {THREE.DefaultLoadingManager} [manager] Extension of {@link THREE.DefaultLoadingManager}
+ */
+THREE.OBJLoader2 = (function () {
+
+	function OBJLoader2( manager ) {
+		this.manager = ( manager == null ) ? THREE.DefaultLoadingManager : manager;
+
+		this.path = '';
+		this.fileLoader = new THREE.FileLoader( this.manager );
+
+		this.meshCreator = new THREE.OBJLoader2.MeshCreator();
+		this.parser = new THREE.OBJLoader2.Parser( this.meshCreator );
```

Comment:
`Parser` is also used by `WWOBJLoader2` to construct the web worker. [WWOBJLoader2.prototype._buildWebWorkerCode](https://github.com/kaisalmen/three.js/blob/WWOBJLoader2/examples/js/loaders/WWOBJLoader2.js#L516-L872) builds a string, that is afterwards put into a blob to create the worker. It basically requires these classes ([link to code](https://github.com/kaisalmen/three.js/blob/WWOBJLoader2/examples/js/loaders/WWOBJLoader2.js#L854-L867)), whereas the top block is **common** code for `OBJLoader2` and `WWOBJLoader2` and the bottom block is defined privately in `WWOBJLoader2`. I didn't want to copy the code into both packages and therefore `Parser` is public:
```
// parser re-construction
this.workerCode += 'if ( THREE.OBJLoader2 === undefined ) { THREE.OBJLoader2 = {} }\n\n';
this.workerCode += buildObject( 'THREE.OBJLoader2.consts', THREE.OBJLoader2.consts );
this.workerCode += buildSingelton( 'THREE.OBJLoader2.Parser', 'Parser', THREE.OBJLoader2.Parser );
this.workerCode += buildSingelton( 'THREE.OBJLoader2._RawObject', '_RawObject', THREE.OBJLoader2._RawObject );
this.workerCode += buildSingelton( 'THREE.OBJLoader2.RawObjectDescription', 'RawObjectDescription', THREE.OBJLoader2.RawObjectDescription );

// web worker construction
this.workerCode += buildSingelton( 'THREE.OBJLoader2.WW.OBJLoader', 'OBJLoader', wwDef );
this.workerCode += buildSingelton( 'THREE.OBJLoader2.WW.MeshCreator', 'MeshCreator', wwMeshCreatorDef );
this.workerCode += 'THREE.OBJLoader2.WW.OBJLoaderRef = new THREE.OBJLoader2.WW.OBJLoader();\n\n';
this.workerCode += buildSingelton( 'THREE.OBJLoader2.WW.OBJLoaderRunner', 'OBJLoaderRunner', wwLoaderRunnerDef );
this.workerCode += 'new THREE.OBJLoader2.WW.OBJLoaderRunner();\n\n';
```

`MeshCreator` on the other hand could be private as the worker has its own MeshCreator that basically supplies transferables back to `WWOBJLoader2` which then creates the Meshes integrated into the scene graph. Initially, exposing `MeshCreator` seemed required, but this is no longer the case.

I could update the code in the dev repo and push updated bundles to this branch.
What do you think?